ufl/algorithms/compute_form_data.py

```diff
@@ -242,11 +246,25 @@ def compute_form_data(form,
     # Note: Default behaviour here will process form the way that is
     # currently expected by vanilla FFC
 
+    # Check that the form does not try to compare complex quantities:
+    # if the quantites being compared are 'provably' real, wrap them
+    # with Real, otherwise throw an error. 
+    # Optimises complex nodes by removing e.g. Conj(Conj(...))
+    if complex_mode:
+        form = do_comparison_check(form)
```

Comment:
OK, I see `do_comparison_check` is actually doing two things.

1. checking for invalid comparisons
2. Marking subtrees as provably `Real`.

Presumably the latter is required for some simplifications?

Reply:
We argue the latter is required to generate valid code (because otherwise you generate a cmp b with a and/or b of type complex (even if of real value) which must be invalid).

Reply:
OK, so can we think of a better name?  The current one suggests that the return-value is unchanged from the input.

Reply:
`sanitize_comparisons`? (Does UFL follow the -ise or -ize convention?)